Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43530&attempt_id=d8247f73-9bfb-4d3c-8dc9-cbd9a9b1e9a3)

Attempted: bazel

<details><summary>Partial log (click to expand)</summary><p>

```
src/main/cpp/blaze_util_posix.cc: In function 'uint64_t blaze::AcquireLock(const string&, bool, bool, blaze::BlazeLock*)':
src/main/cpp/blaze_util_posix.cc:667:3: warning: ignoring return value of 'int ftruncate(int, __off_t)', declared with attribute warn_unused_result [-Wunused-result]
   (void) ftruncate(lockfd, 0);
   ^~~~~~~~~~~~~~~~~~~~~~~~~~~
src/main/cpp/blaze_util_posix.cc: At global scope:
src/main/cpp/blaze_util_posix.cc:404:15: warning: 'std::__cxx11::string blaze::RunProgram(const string&, const std::vector<std::__cxx11::basic_string<char> >&)' defined but not used [-Wunused-function]
 static string RunProgram(const string& exe,
               ^~~~~~~~~~
building of ‘/nix/store/hbcq8a1nkqi509skcjdpck4di0cyn9n8-bazel-0.13.0.drv’ timed out after 1800 seconds
error: build of ‘/nix/store/hbcq8a1nkqi509skcjdpck4di0cyn9n8-bazel-0.13.0.drv’ failed
```
</p></details>

--- a/pkgs/development/tools/build-managers/bazel/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/default.nix
@@ -96,6 +96,15 @@ stdenv.mkDerivation rec {
     sed -i -e "361 a --host_copt=\"$(echo $NIX_CFLAGS_COMPILE | sed -e 's/ /" --host_copt=\"/g')\" \\\\" scripts/bootstrap/compile.sh
     sed -i -e "361 a --linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --linkopt=\"-Wl,/g')\" \\\\" scripts/bootstrap/compile.sh
     sed -i -e "361 a --host_linkopt=\"-Wl,$(echo $NIX_LDFLAGS | sed -e 's/ /" --host_linkopt=\"-Wl,/g')\" \\\\" scripts/bootstrap/compile.sh
+
+    # --experimental_strict_action_env (which will soon become the
+    # default, see bazelbuild/bazel#2574) hardcodes the default
+    # action environment to a value that on NixOS at least is bogus.
+    # So we hardcode it to something useful.
+    substituteInPlace \
+		  src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRuleClassProvider.java \
+      --replace /bin:/usr/bin ${defaultShellPath}
+
     patchShebangs .
   '';
 

--- a/pkgs/development/tools/build-managers/bazel/default.nix
+++ b/pkgs/development/tools/build-managers/bazel/default.nix
@@ -1,4 +1,6 @@
-{ stdenv, lib, fetchurl, runCommand, jdk, zip, unzip, bash, writeCBin, coreutils, makeWrapper, which, python, gnused
+{ stdenv, lib, fetchurl, runCommand, makeWrapper
+, jdk, zip, unzip, bash, writeCBin, coreutils
+, which, python, gnused, gnugrep, findutils
 # Always assume all markers valid (don't redownload dependencies).
 # Also, don't clean up environment variables.
 , enableNixHacks ? false
@@ -6,16 +8,21 @@
 , libcxx, CoreFoundation, CoreServices, Foundation
 }:
 
-let srcDeps = stdenv.lib.singleton (
-      fetchurl {
-        url = "https://github.com/google/desugar_jdk_libs/archive/f5e6d80c6b4ec6b0a46603f72b015d45cf3c11cd.zip";
-        sha256 = "c80f3f3d442d8a6ca7adc83f90ecd638c3864087fdd6787ffac070b6f1cc8f9b";
-      }
-    );
-    distDir = runCommand "bazel-deps" {} ''
-      mkdir -p $out
-      for i in ${builtins.toString srcDeps}; do cp $i $out/$(stripHash $i); done
-    '';
+let
+  srcDeps = stdenv.lib.singleton (
+    fetchurl {
+      url = "https://github.com/google/desugar_jdk_libs/archive/f5e6d80c6b4ec6b0a46603f72b015d45cf3c11cd.zip";
+      sha256 = "c80f3f3d442d8a6ca7adc83f90ecd638c3864087fdd6787ffac070b6f1cc8f9b";
+    }
+  );
+
+  distDir = runCommand "bazel-deps" {} ''
+    mkdir -p $out
+    for i in ${builtins.toString srcDeps}; do cp $i $out/$(stripHash $i); done
+  '';
+
+  defaultShellPath = lib.makeBinPath [ bash coreutils findutils gnugrep gnused which ];
+
 in
 stdenv.mkDerivation rec {
 
@@ -52,7 +59,7 @@ stdenv.mkDerivation rec {
 
     int main(int argc, char *argv[]) {
       char *path = getenv("PATH");
-      char *pathToAppend = "${lib.makeBinPath [ coreutils gnused ]}";
+      char *pathToAppend = "${defaultShellPath}";
       char *newPath;
       if (path != NULL) {
         int length = strlen(path) + 1 + strlen(pathToAppend) + 1;
@@ -140,7 +147,7 @@ stdenv.mkDerivation rec {
   # Save paths to hardcoded dependencies so Nix can detect them.
   postFixup = ''
     mkdir -p $out/nix-support
-    echo "${customBash} ${gnused} ${coreutils}" > $out/nix-support/depends
+    echo "${customBash} ${defaultShellPath}" > $out/nix-support/depends
   '';
 
   dontStrip = true;

